models/qwen3/14b/decode_layer_a8w8.py

```
@@ -0,0 +1,1596 @@
+# dim, so this only sizes the test fixtures.
+MAX_BLOCKS_PER_SEQ = MAX_CTX_BLOCKS
+NUM_PAGES = BATCH * MAX_BLOCKS_PER_SEQ
+CACHE_ROWS = NUM_PAGES * NUM_KV_HEADS * BLOCK_SIZE  # paged k_cache / v_cache rows (one layer)
+
+# ── Scope 2 · KV-block split (flash-decoding) for ragged load balance ──
+# Each fa_fused lane (a Q-group pair) is split into contiguous KV partitions of
+# TOKENS_PER_SPLIT tokens. Smaller TOKENS_PER_SPLIT = finer load balance for
+# ragged seq_lens, at the cost of more fa_fused tasks
+# (BATCH * (NUM_KV_HEADS // 2) * KV_SPLITS).
+# Dispatch unit = ONE seq block (TOKENS_PER_SPLIT == SEQ_TILE). Every fa_fused
+# work item is then a single SEQ_TILE block (×2 heads) — equal cost regardless of
```

Ruff (0.15.18)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)